pkg/apis/autoscaling/v2beta2/conversion.go

```diff
@@ -27,16 +30,74 @@ func Convert_autoscaling_HorizontalPodAutoscaler_To_v2beta2_HorizontalPodAutosca
 	if err := autoConvert_autoscaling_HorizontalPodAutoscaler_To_v2beta2_HorizontalPodAutoscaler(in, out, s); err != nil {
 		return err
 	}
-	// v2beta2 round-trips to internal without any serialized annotations, make sure any from other versions don't get serialized
-	out.Annotations, _ = autoscaling.DropRoundTripHorizontalPodAutoscalerAnnotations(out.Annotations)
+	// Ensure old round-trips annotations are discarded
+	annotations, copiedAnnotations := autoscaling.DropRoundTripHorizontalPodAutoscalerAnnotations(out.Annotations)
+	out.Annotations = annotations
+
+	behavior := in.Spec.Behavior
```

Comment:
I'm a bit confused here because this new field is nested under behavior...

Wouldn't it get serialized to the behavior annotation?

Reply:
Resolved.